pkg/services/ngalert/api/util.go

```diff
@@ -85,9 +87,15 @@ type AlertingProxy struct {
 // withReq proxies a different request
 func (p *AlertingProxy) withReq(
 	ctx *models.ReqContext,
-	req *http.Request,
+	method string,
+	u *url.URL,
+	body io.Reader,
 	extractor func([]byte) (interface{}, error),
 ) response.Response {
+	req, err := http.NewRequest(method, u.String(), body)
+	if err != nil {
+		return response.Error(400, err.Error(), nil)
```

Comment:
Why do we have hardcoded `400 status` here? Shouldn't we return the status from the request?

Reply:
At this point the request has not run yet (it runs [here](https://github.com/grafana/grafana/blob/3cb4904fe89c2a4ac8b89a360b11fadd7e623e41/pkg/services/ngalert/api/util.go#L101)); this catches the case when for some reason the request object has failed to be created.
But I agree that it's better to use `http.StatusBadRequest` instead of 400